pkg/api/v1alpha2/api.proto

```diff
@@ -0,0 +1,481 @@
+/**
+ * Katib API
+ */
+syntax = "proto3";
+
+package api.v1.alpha2;
+
+import "google/api/annotations.proto";
+
+/**
+ * Service for Main API for Katib
+ * For each RPC service, we define mapping to HTTP REST API method.
+ * The mapping includes the URL path, query parameters and request body.
+ * https://cloud.google.com/service-infrastructure/docs/service-management/reference/rpc/google.api#http
+ */
+service Manager {
+    /**
+     * Register a Experiment to DB.
+     */
+    rpc RegisterExperiment(RegisterExperimentRequest) returns (RegisterExperimentReply){
+        option (google.api.http) = {
+            post: "/api/Manager/RegisterExperiment"
+            body: "experiment"
+        };
+    };
+
+    /** 
+     * Delete a Experiment from DB by name.
+     */
+    rpc DeleteExperiment(DeleteExperimentRequest) returns (DeleteExperimentReply){
+        option (google.api.http) = {
+            get: "/api/Manager/DeleteExperiment/{experiment_name}"
+        };
+    };
+
+    /** 
+     * Get a Experiment from DB by name.
+     */
+    rpc GetExperiment(GetExperimentRequest) returns (GetExperimentReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetExperiment/{experiment_name}"
+        };
+    };
+
+    /**
+     * Get a summary list of Experiment from DB.
+     * The summary includes name and condition.
+     */
+    rpc GetExperimentList(GetExperimentListRequest) returns (GetExperimentListReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetExperimentList"
+        };
+    };
+
+    /** 
+     * Update Status of a experiment.
+     */
+    rpc UpdateExperimentStatus(UpdateExperimentStatusRequest) returns (UpdateExperimentStatusReply){
```

Comment:
Currently, Suggestion algorithms can modify the algorithm parameters. https://github.com/kubeflow/katib/blob/master/pkg/suggestion/grid_service.go#L195 

How is this handled?

Reply:
I will create UpdateAlgorithmParameter API.